src/Compiler/lex.fsl

```diff
@@ -1427,14 +1527,137 @@ and tripleQuoteString sargs skip = parse
 
  | eof
     { let (_buf, _fin, m, kind, args) = sargs
-      EOF (LexCont.String(args.ifdefStack, args.stringNest, LexerStringStyle.TripleQuote, kind, m)) }
+      EOF (LexCont.String(args.ifdefStack, args.stringNest, LexerStringStyle.TripleQuote, kind, args.interpolationDelimiterLength, m)) }
+
+ | surrogateChar surrogateChar // surrogate code points always come in pairs
+ | _
+    { let (buf, _fin, m, kind, args) = sargs
+      addUnicodeString buf (lexeme lexbuf)
+      if not skip then
+          STRING_TEXT (LexCont.String(args.ifdefStack, args.stringNest, LexerStringStyle.TripleQuote, kind, args.interpolationDelimiterLength, m))
+      else
+          tripleQuoteString sargs skip lexbuf }
+
+and extendedInterpolatedString sargs skip = parse
+ |  '"' '"' '"'
+    { let (buf, fin, _m, kind, args) = sargs
+      args.interpolationDelimiterLength <- 0
+      let cont = LexCont.Token(args.ifdefStack, args.stringNest)
+      fin.Finish buf kind LexerStringFinisherContext.TripleQuote cont }
+
+ | newline
+    { let (buf, _fin, m, kind, args) = sargs
+      newline lexbuf
+      addUnicodeString buf (lexeme lexbuf)
+      if not skip then
+          STRING_TEXT (LexCont.String(args.ifdefStack, args.stringNest, LexerStringStyle.ExtendedInterpolated, kind, args.interpolationDelimiterLength, m))
+      else
+          extendedInterpolatedString sargs skip lexbuf }
+
+// The rest is to break into pieces to allow double-click-on-word and other such things
```

Comment:
clarify

Reply:
I am not entirely sure, so I will leave it as is (this part is the same as in the pre-existing `tripleQuoteString` lexer rule)